### PR TITLE
support pp sharding stage 1 overlap

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Description
support pp sharding stage 1 overlap

### CODE REVIEW

File: paddlenlp/trainer/trainer.py [hunk 1]
```diff
@@ -1620,6 +1620,7 @@ def _prepare_training(self, data):
         inputs = _prepare_training(model, inputs)
```
Comment (ZHUI) [inline paddlenlp/trainer/trainer.py]: https://github.com/PaddlePaddle/Paddle/blob/7472057c9884b6c1b1030b322eb3d89d719aff77/python/paddle/distributed/fleet/meta_parallel/pipeline_parallel.py#L560C1-L564C1 

这里 重写了_prepare_training ，要不直接这里加几行？

Comment (FeixLiu) [reply paddlenlp/trainer/trainer.py]: done

**Review by ZHUI — APPROVED** [review]
LGTM

**Review by ZHUI — APPROVED** [review]
LGTM

**Pull request merged**